src/CommunityToolkit.Maui.MediaElement/PlatformView/MediaManager.macios.cs

```diff
@@ -0,0 +1,459 @@
+﻿using AVFoundation;
+using AVKit;
+using CoreFoundation;
+using CoreMedia;
+using Foundation;
+using Microsoft.Extensions.Logging;
+
+namespace CommunityToolkit.Maui.MediaElement;
+
+public partial class MediaManager : IDisposable
+{
+	const NSKeyValueObservingOptions valueObserverOptions =
+		NSKeyValueObservingOptions.Initial | NSKeyValueObservingOptions.New;
+
+	NSObject? playedToEndObserver;
+	NSObject? itemFailedToPlayToEndTimeObserver;
+	NSObject? playbackStalledObserver;
+	NSObject? errorObserver;
+	IDisposable? statusObserver;
+	IDisposable? volumeObserver;
+	IDisposable? rateObserver;
+	IDisposable? timeControlStatusObserver;
+	IDisposable? currentItemErrorObserver;
+	AVPlayerViewController? playerViewController;
+	AVPlayerItem? playerItem;
```

Comment:
having those `private` will block devs to customize the MediaManager, I suggest to change it to `protected`